src/main/kotlin/com/parmet/buf/gradle/GenerateConfiguration.kt

```diff
@@ -23,15 +23,10 @@ const val BUF_GENERATE_TASK_NAME = "bufGenerate"
 const val GENERATED_DIR = "generated"
 
 internal fun Project.configureGenerate() {
-    if (hasGenerate()) {
```

Comment:
It's worth noting that with this check, the task never gets a chance to override the gen file name/location. If there is a better way to perform the check I am happy to update the PR.

Reply:
Can the check be preserved by looking for either of `buf.gen.yaml` or the override file?

There are a few validation cases we should add as well:
- Validate that both `buf.gen.yaml` and a file override are not both present and fail if that is true.
- Validate that the chosen file exists and that it is a file rather than just conditionally set up the task as is done today - fail fast.

Reply:
In my testing, if I had an override specified in the gradle file, I couldn't read that quite yet as the generateTask had not been registered/configured yet (somewhat of a chicken and egg problem). Is there a way to read the task configuration ahead of time? If so, then we can easily add both those checks.

Reply:
The checks can come at task run time instead.

Reply:
Ok, I've added `configureCheckBufGenTask` which is triggered after evaluation. This is a default Gradle task that checks the aforementioned criteria.

Reply:
I don't think it should be a second task - can it run as the first action of the `bufGenerate` task as part of config file resolution? This is how `bufLint` does it for `buf.yaml`, for example.

This validation should also be tested - `buf.yaml` tests look like https://github.com/andrewparmet/buf-gradle-plugin/blob/4c1f4287d68aa6a04a1979dbdf86abf0b8775776/src/test/resources/LintTest/lint%20with%20a%20file%20and%20dependency%20config%20override%20fails/build.gradle.

Reply:
I've gone ahead and migrated the validation logic to be a part of the config file resolution in bufGenerate. Note that we will always specify `template-name` even in the case of the root buf.gen.yaml. This is innocuous, but I thought should mention that.

Reply:
I think that's fine.